src/Libraries/Microsoft.Extensions.Telemetry/Http/HttpRouteParser.cs

```diff
@@ -187,13 +187,42 @@ private static Segment GetParameterSegment(string httpRoute, ref int pos)
                 }
             }
 
+            // The segment has '*' catch all parameter.
+            // When we meet the character it indicates param start position needs to be adjusted, so that we capture 'param' instead of '*param'
+            // *param can only appear after opening curly brace and position needs to be adjusted only once
+            else if (!catchAllParamFound && ch == '*' && pos > 0 && httpRoute[pos - 1] == '{')
+            {
+                paramNameStart++;
+
+                // Catch all parameters can start with one or two '*' characters.
+                if (httpRoute[paramNameStart] == '*')
```

Comment:
Do we need to validate `paramNameStart` here against `httpRoute` length?
I'm thinking of ill-formed strings, not sure if there's any logic upstream that prevents that from ever happening.

Reply:
In general, malformed routes are not handled properly in this parser, causing IndexOutOfRange exceptions. It's a bigger issue than just his part. I would consider doing this in a separate PR because of the refactoring required.

Reply:
For incoming requests, the route template is statically checked by analyzers and issue diagnostic warnings when it's not correct. However, for outgoing requests using downstream dependency metadata route info, there are no such analyzers.